comp/dogstatsd/statsd/statsd_otel.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build otlp
+
+package statsd
+
+import (
+	"go.uber.org/fx"
+
+	ddgostatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/metricsclient"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// OTelStatsdModule defines the fx options for the OTel statsd otelcomponent.
+// This should only be used in the OTel agent.
+func OTelStatsdModule() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newOTelStatsdComp))
+}
+
+type otelcomponent struct {
+	client ddgostatsd.ClientInterface
+}
+
+func newOTelStatsdComp() Component {
+	return &otelcomponent{metricsclient.NewStatsdClientWrapper(&ddgostatsd.NoOpClient{})}
+}
+
+// Get returns a pre-configured and shared statsd client (requires STATSD_URL env var to be set)
+func (m *otelcomponent) Get() (ddgostatsd.ClientInterface, error) {
+	return m.client, nil
+}
+
+// Create returns a pre-configured statsd client
+func (m *otelcomponent) Create(_ ...ddgostatsd.Option) (ddgostatsd.ClientInterface, error) {
+	return m.client, nil
+}
+
```

Comment:
I don't see any value for this module definition.  We can create an instance of StatsdClientWrapper in run function and provide it as an statsd component. We do something similar here https://github.com/DataDog/datadog-agent/blob/ac20d1cf233bcac3a2de8a7100170c0fa46ba7cf/cmd/otel-agent/subcommands/run/command.go#L124-L131
This way we can avoid cast in collector.go

Reply:
This is not only used in otel, it is also used in trace agent component. Trace agent requires a statsd component rather than a statsd client instance https://github.com/DataDog/datadog-agent/blob/main/comp/trace/agent/impl/agent.go#L63. So we must have a statsd component and provide it in fx.

Reply:
and statsd component needs to implement all these interfaces: https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/statsd/component.go#L16

Reply:
Thanks, makes sense. In this I would recommend to refactor the code where FX call `NewStatsdClientWrapper` to create wrapper and provide it as `ClientInterface` to `OTelStatsdModule` and `StatsdClientWrapper` to the collector.

Reply:
OK makes sense, made the change in https://github.com/DataDog/datadog-agent/pull/26945/commits/ceb8565888dd61b3b468367cca59583297f27453